File: app/src/main/java/com/manrique/trailerstock/dao/CategoriaDao.java
package com.manrique.trailerstock.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import com.manrique.trailerstock.model.Categoria;
import java.util.List;

@Dao
public interface CategoriaDao {

    @Insert
    void insertar(Categoria categoria);

    @Update
    void actualizar(Categoria categoria);

    @Delete
    void eliminar(Categoria categoria);

    @Query("SELECT * FROM categorias ORDER BY nombre ASC")
    LiveData<List<Categoria>> getAllCategorias();

    @Query("SELECT * FROM categorias ORDER BY nombre ASC")
    List<Categoria> obtenerTodas();

    @Query("SELECT * FROM categorias WHERE id = :id")
    Categoria obtenerPorId(int id);
}

File: app/src/main/java/com/manrique/trailerstock/data/local/dao/CategoriaDao.kt
package com.manrique.trailerstock.data.local.dao

import androidx.room.*
import com.manrique.trailerstock.data.local.entities.Categoria
import kotlinx.coroutines.flow.Flow

/**
 * DAO para operaciones de Categoria con soporte de Coroutines.
 */
@Dao
interface CategoriaDao {
    
    @Query("SELECT * FROM categorias WHERE eliminado = 0 ORDER BY nombre ASC")
    fun obtenerTodas(): Flow<List<Categoria>>
    
    @Query("SELECT * FROM categorias WHERE id = :id")
    suspend fun obtenerPorId(id: Int): Categoria?
    
    @Query("UPDATE categorias SET eliminado = 1 WHERE id = :id")
    suspend fun softDelete(id: Int)
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertar(categoria: Categoria): Long
    
    @Update
    suspend fun actualizar(categoria: Categoria)
    
    @Delete
    suspend fun eliminar(categoria: Categoria)
    
    @Query("SELECT COUNT(*) FROM categorias")
    suspend fun contar(): Int
}
